openlabcmd/openlabcmd/cli.py

```diff
@@ -214,6 +215,11 @@ def check(self):
                 # the failed flag would be record when do check()
                 if self.args.recover and plugin.failed:
                     plugin.recover()
+                if plugin.failed:
+                    exit_flag = 1
+
+        if not exit_flag:
+            exit(exit_flag)
```

Comment:
Instead of `exit`, I suggest to raise ClientException error like other commands do to keep the same behavior.

Reply:
Good idea, but in current implements, if we `raise ClientException`, the exit errno is still zero. because the "self.run" is not returned. We need change it to:
```python
    def _main(self):
        # ... ...
        return self.run()
```

Does it also have some impact on ha work? If not, I can address it in next PS.

[1] https://github.com/theopenlab/labkeeper/blob/0423574/openlabcmd/openlabcmd/cli.py#L339

Reply:
do it!

Reply:
done